Formula/r/recc.rb

```diff
@@ -0,0 +1,231 @@
+
+    generate_recc_server bin
+
+    generate_recc_conf prefix / "etc/recc/"
+  end
+
+  def macos_caveats
+    on_macos do
+      command_output = Utils.safe_popen_read("launchctl", "limit", "maxfiles")
+      maxfiles_limit = command_output.split[1].to_i
+      ulimit_sn = Utils.safe_popen_read("ulimit", "-Sn").to_i
+
+      general_info = <<~EOS
+        If you are starting the `recc` service via `brew service start`, please ensure the maxfiles limit
+        for `launchctl` is larger than 364. The curent limit is: #{maxfiles_limit}. If you are running
+        `recc-server` directly, please ensure `ulimit -Sn` is larger than 364. The current limit is #{ulimit_sn}.
+      EOS
+
+      if maxfiles_limit < 365
+        launchctl_help = <<~EOS
+
+          To increase the limit and make the change persist across reboots, you can create a launch daemon
+          using the following steps:
+
+          sudo /usr/libexec/PlistBuddy /Library/LaunchDaemons/limit.maxfiles.plist \\
+            -c "add Label string com.launchd.maxfiles" \\
+            -c "add ProgramArguments array" \\
+            -c "add ProgramArguments: string launchctl" \\
+            -c "add ProgramArguments: string limit" \\
+            -c "add ProgramArguments: string maxfiles" \\
+            -c "add ProgramArguments: string 512" \\
+            -c "add ProgramArguments: string unlimited" \\
+            -c "add RunAtLoad bool true" && \\
+          sudo chown root:wheel /Library/LaunchDaemons/limit.maxfiles.plist && \\
+          sudo chmod 644 /Library/LaunchDaemons/limit.maxfiles.plist && \\
+          sudo launchctl load -w /Library/LaunchDaemons/limit.maxfiles.plist#{" "}
+        EOS
+      else
+        launchctl_help = ""
+      end
+      (general_info + launchctl_help)
+    end
+  end
+
+  def caveats
+    <<~EOS
+      The recc configuration is read in the following order.
+        1. Environment variables (see: https://buildgrid.gitlab.io/recc/configuration-variables.html)
+        2. $(pwd)/recc/recc.conf
+        3. {$RECC_CONFIG_DIRECTORY}/recc.conf
+        4. ~/.recc/recc.conf
+        5. #{etc}/recc/recc.conf
+        6. #{bin}/../etc/recc/recc.conf
+      If an option is set in multiple places, the one higher up in the list will be effective.
+
+      To launch a compiler with recc, set the following variables:
+        CC=#{bin}/recc-cc
+        CXX=#{bin}/recc-c++
+
+      For CMake builds, an alternative option is to use CMAKE_<LANG>_COMPILER/LINKER_LAUNCHER.
+      Please do not set both CC/CXX and CMAKE_<LANG>COMPILER?LINKER_LAUNCHER variables to recc.
+        CMAKE_C_COMPILER_LAUNCHER=#{bin}/recc
+        CMAKE_CXX_COMPILER_LAUNCHER=#{bin}/recc
+        CMAKE_C_LINKER_LAUNCHER=#{bin}/recc
+        CMAKE_CXX_LINKER_LAUNCHER=#{bin}/recc
+
+      Start the recc service to use the out-of-the-box configuration (see instructions below).
+      #{macos_caveats}
+    EOS
+  end
+
+  test do
+    system bin/"recc", "--version"
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Updated to test the feature.